block/manager.go

```diff
@@ -65,15 +64,15 @@ type Manager struct {
 	// Protect against processing two blocks at once when there are two routines handling incoming gossiped blocks,
 	// and incoming DA blocks, respectively.
 	retrieverMu sync.Mutex
-	Retriever   da.BatchRetriever
-	// get the next target height to sync local state to
-	targetSyncHeight diodes.Diode
-	// TargetHeight holds the value of the current highest block seen from either p2p (probably higher) or the DA
-	TargetHeight atomic.Uint64
-
+	// Protect against syncing twice from DA in case new batch is posted but it did not finish to sync yet.
+	syncFromDaMu sync.Mutex
+	Retriever    da.BatchRetriever
 	// Cached blocks and commits for applying at future heights. The blocks may not be valid, because
 	// we can only do full validation in sequential order.
 	blockCache *Cache
+
+	// TargetHeight holds the value of the current highest block seen from either p2p (probably higher) or the DA
+	TargetHeight atomic.Uint64
```

Comment:
comment is wrong now since also for settlement

Reply:
can also check relevance of this issue? https://github.com/dymensionxyz/dymint/issues/987

Reply:
settlement and DA is the same. and the comment is not from this pr